@clalancette the windows build is failing. I tried to reproduce this locally,

if I use `ament_cmake` and set manually `add_definitions(-DROS_PACKAGE_NAME=\"${PROJECT_NAME}\")` The compilation is fine, 

but with `ament_cmake_ros` I'm getting these errors:

```
16:28:24 LINK : fatal error LNK1181: cannot open input file 'Release\rviz_rendering_test_utils.lib' [C:\ci\ws\build\rviz_rendering\billboard_line_test_target.vcxproj]

16:28:24   Building Custom Rule C:/ci/ws/src/ros2/rviz/rviz_rendering/CMakeLists.txt

16:28:24   covariance_visual_test.cpp

16:28:24 LINK : fatal error LNK1181: cannot open input file 'Release\rviz_rendering_test_utils.lib' [C:\ci\ws\build\rviz_rendering\covariance_visual_test_target.vcxproj]

16:28:24   Building Custom Rule C:/ci/ws/src/ros2/rviz/rviz_rendering/CMakeLists.txt
```

any thoughts?

--- a/rviz_rendering/CMakeLists.txt
+++ b/rviz_rendering/CMakeLists.txt
@@ -165,10 +165,12 @@ if(BUILD_TESTING)
   find_package(ament_cmake_gmock REQUIRED)
   find_package(ament_cmake_gtest REQUIRED)
 
-  add_library(rviz_rendering_test_utils
+  add_library(rviz_rendering_test_utils SHARED
     test/rviz_rendering/ogre_testing_environment.cpp
     test/rviz_rendering/scene_graph_introspection.cpp
   )
+  target_compile_definitions(rviz_rendering_test_utils PRIVATE
+  "TEST_RVIZ_RENDERING_BUILDING_LIBRARY")
   target_link_libraries(rviz_rendering_test_utils
     PUBLIC
     rviz_ogre_vendor::OgreMain

--- a/rviz_rendering/CMakeLists.txt
+++ b/rviz_rendering/CMakeLists.txt
@@ -165,12 +165,12 @@ if(BUILD_TESTING)
   find_package(ament_cmake_gmock REQUIRED)
   find_package(ament_cmake_gtest REQUIRED)
 
-  add_library(rviz_rendering_test_utils SHARED
+  add_library(rviz_rendering_test_utils
     test/rviz_rendering/ogre_testing_environment.cpp
     test/rviz_rendering/scene_graph_introspection.cpp
   )
   target_compile_definitions(rviz_rendering_test_utils PRIVATE
-  "TEST_RVIZ_RENDERING_BUILDING_LIBRARY")
+    "TEST_RVIZ_RENDERING_BUILDING_LIBRARY")
   target_link_libraries(rviz_rendering_test_utils
     PUBLIC
     rviz_ogre_vendor::OgreMain

--- a/rviz_rendering/test/rviz_rendering/ogre_testing_environment.hpp
+++ b/rviz_rendering/test/rviz_rendering/ogre_testing_environment.hpp
@@ -30,6 +30,8 @@
 #ifndef RVIZ_RENDERING__OGRE_TESTING_ENVIRONMENT_HPP_
 #define RVIZ_RENDERING__OGRE_TESTING_ENVIRONMENT_HPP_
 
+#include "./visibility_control.hpp"
+
 namespace rviz_rendering
 {
 class OgreTestingEnvironment
@@ -41,12 +43,16 @@ public:
    * @param: bool debug, if true, all logging of Ogre is send to std::out, if false no logging
    * occurs. Since the logging pollutes the test output, it defaults to false
    */
+  TEST_RVIZ_RENDERING_PUBLIC
   void setUpOgreTestEnvironment(bool debug = false);
 
+  TEST_RVIZ_RENDERING_PUBLIC
   void setUpRenderSystem();
 
+  TEST_RVIZ_RENDERING_PUBLIC
   int getGlslVersion() const;
 
+  TEST_RVIZ_RENDERING_PUBLIC
   ~OgreTestingEnvironment();
 };
 

--- a/rviz_rendering/test/rviz_rendering/scene_graph_introspection.hpp
+++ b/rviz_rendering/test/rviz_rendering/scene_graph_introspection.hpp
@@ -45,23 +45,35 @@
 #include "rviz_rendering/objects/point_cloud.hpp"
 #include "rviz_rendering/objects/movable_text.hpp"
 
+#include "./visibility_control.hpp"
+
 namespace rviz_rendering
 {
+TEST_RVIZ_RENDERING_PUBLIC
 std::vector<Ogre::SceneNode *> findAllArrows(Ogre::SceneNode * scene_node);
 
+TEST_RVIZ_RENDERING_PUBLIC
 std::vector<Ogre::SceneNode *> findAllAxes(Ogre::SceneNode * scene_node);
 
+TEST_RVIZ_RENDERING_PUBLIC
 std::vector<Ogre::Entity *> findAllEntitiesByMeshName(
   Ogre::SceneNode * scene_node, const Ogre::String & resource_name);
+TEST_RVIZ_RENDERING_PUBLIC
 Ogre::Entity * findEntityByMeshName(
   Ogre::SceneNode * scene_node, const Ogre::String & resource_name);
 
+TEST_RVIZ_RENDERING_PUBLIC
 std::vector<Ogre::Entity *> findAllSpheres(Ogre::SceneNode * scene_node);
+TEST_RVIZ_RENDERING_PUBLIC
 std::vector<Ogre::Entity *> findAllCones(Ogre::SceneNode * scene_node);
+
+TEST_RVIZ_RENDERING_PUBLIC
 std::vector<Ogre::Entity *> findAllCylinders(Ogre::SceneNode * scene_node);
 
+TEST_RVIZ_RENDERING_PUBLIC
 std::vector<rviz_rendering::PointCloud *> findAllPointClouds(Ogre::SceneNode * scene_node);
 
+TEST_RVIZ_RENDERING_PUBLIC
 Ogre::BillboardChain * findOneBillboardChain(Ogre::SceneNode * scene_node);
 
 template<typename OgreType>

--- a/rviz_rendering/test/rviz_rendering/visibility_control.hpp
+++ b/rviz_rendering/test/rviz_rendering/visibility_control.hpp
@@ -1,0 +1,58 @@
+// Copyright 2023 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TEST_RVIZ_RENDERING__VISIBILITY_CONTROL_HPP_  // NOLINT
+#define TEST_RVIZ_RENDERING__VISIBILITY_CONTROL_HPP_  // NOLINT
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+// This logic was borrowed (then namespaced) from the examples on the gcc wiki:
+//     https://gcc.gnu.org/wiki/Visibility
+
+#if defined _WIN32 || defined __CYGWIN__
+  #ifdef __GNUC__
+    #define TEST_RVIZ_RENDERING_EXPORT __attribute__ ((dllexport))
+    #define TEST_RVIZ_RENDERING_IMPORT __attribute__ ((dllimport))
+  #else
+    #define TEST_RVIZ_RENDERING_EXPORT __declspec(dllexport)
+    #define TEST_RVIZ_RENDERING_IMPORT __declspec(dllimport)
+  #endif
+  #ifdef TEST_RVIZ_RENDERING_BUILDING_LIBRARY
+    #define TEST_RVIZ_RENDERING_PUBLIC TEST_RVIZ_RENDERING_EXPORT
+  #else
+    #define TEST_RVIZ_RENDERING_PUBLIC TEST_RVIZ_RENDERING_IMPORT
+  #endif
+  #define TEST_RVIZ_RENDERING_PUBLIC_TYPE TEST_RVIZ_RENDERING_PUBLIC
+  #define TEST_RVIZ_RENDERING_LOCAL
+#else
+  #define TEST_RVIZ_RENDERING_EXPORT __attribute__ ((visibility("default")))
+  #define TEST_RVIZ_RENDERING_IMPORT
+  #if __GNUC__ >= 4
+    #define TEST_RVIZ_RENDERING_PUBLIC __attribute__ ((visibility("default")))
+    #define TEST_RVIZ_RENDERING_LOCAL  __attribute__ ((visibility("hidden")))
+  #else
+    #define TEST_RVIZ_RENDERING_PUBLIC
+    #define TEST_RVIZ_RENDERING_LOCAL
+  #endif
+  #define TEST_RVIZ_RENDERING_PUBLIC_TYPE
+#endif
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif  // TEST_RVIZ_RENDERING__VISIBILITY_CONTROL_HPP_  // NOLINT

